test/Base_Tests/src/Data/Decimal_Spec.enso

```diff
@@ -492,6 +499,91 @@ add_specs suite_builder =
             nd2 = -d2
             nd2 . should_equal 5
 
+    suite_builder.group "conversions" group_builder->
+        group_builder.specify "should convert correctly to and from Integer" <|
+            a = Decimal.new "12000"
+            b = Decimal.new "12000" (Math_Context.new 2)
+            c = Decimal.new "12000" (Math_Context.new 3)
+            d = Decimal.new "2345.6"
+            huge_a = Decimal.new "3.4E320"
+            huge_b = Decimal.new "3.400E320"
+            huge_c = Decimal.new "3.4000000E320"
+            huge_integer = 34 * (10 ^ 319)
+
+            a.to_integer . should_equal 12000
+            b.to_integer . should_equal 12000
+            c.to_integer . should_equal 12000
+            d.to_integer . should_equal 2345
+            huge_a.to_integer . should_equal huge_integer
+            huge_b.to_integer . should_equal huge_integer
+            huge_c.to_integer . should_equal huge_integer
+
+            (-a).to_integer . should_equal -12000
+            (-b).to_integer . should_equal -12000
+            (-c).to_integer . should_equal -12000
+            (-d).to_integer . should_equal -2345
+            (-huge_a).to_integer . should_equal -huge_integer
+            (-huge_b).to_integer . should_equal -huge_integer
+            (-huge_c).to_integer . should_equal -huge_integer
+
+        group_builder.specify "should convert correctly to Float" <|
+            a = Decimal.new "12000"
+            b = Decimal.new "12000" (Math_Context.new 2)
+            c = Decimal.new "12000" (Math_Context.new 3)
+            huge_a = Decimal.new "3.4E300"
+            huge_b = Decimal.new "3.400E300"
+            huge_c = Decimal.new "3.4000000E300"
+            huge_float = 3.4 * (10 ^ 300)
+
+            a.to_float . should_equal 12000.0
+            b.to_float . should_equal 12000.0
+            c.to_float . should_equal 12000.0
+            huge_a.to_float . should_equal huge_float
+            huge_b.to_float . should_equal huge_float
+            huge_c.to_float . should_equal huge_float
+
+            (-a).to_float . should_equal -12000
+            (-b).to_float . should_equal -12000
+            (-c).to_float . should_equal -12000
+            (-huge_a).to_float . should_equal -huge_float
+            (-huge_b).to_float . should_equal -huge_float
+            (-huge_c).to_float . should_equal -huge_float
+
+        group_builder.specify "attaches a warning to the result of .to_float when it is out the representable range" <|
+            huge_a = Decimal.new "3.4E320"
+            f = huge_a.to_float
+            f.should_equal Number.positive_infinity
+            Problems.expect_only_warning Out_Of_Range f
+            f2 = (-huge_a).to_float
+            f2.should_equal Number.negative_infinity
+            Problems.expect_only_warning Out_Of_Range f2
```

Comment:
please test the `to_display_text` of the warning.

We often used to have typos inside of `to_display_text`, especially after refacors. If it's not unit tested, we only learn about it when the warning/error comes up in the GUI and is scrambled. Better to have tests for this since we don't have much static analysis capabilities.

Reply:
Done in #9586.